src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -204,7 +242,7 @@ private int LookupAggregatorStore(string[] tagKeys, object[] tagValues, int leng
 
                         ref var metricPoint = ref this.metricPoints[aggregatorIndex];
                         var dt = DateTimeOffset.UtcNow;
-                        metricPoint = new MetricPoint(this.aggType, dt, seqKey, seqVal, this.histogramBounds);
+                        metricPoint = new MetricPoint(this.aggType, MetricPointStatus.UpdatePending, dt, seqKey, seqVal, this.histogramBounds);
```

Comment:
when we re-use a MetricPoint, the MetricPoint will internally create fresh arrays for the BucketCounts. Given the fact that the buckets don't change, we should reuse the same array (after resetting all values to zero).
The arrays can be constructed one time (at AggregatorStore ctor), and reused afterwards. Else, it'd result in a lot of arrays being created/thrown away for histograms.

(will address this separately, once this is done)